geode-core/src/main/java/org/apache/geode/internal/cache/DistributedRegion.java

```diff
@@ -3694,7 +3694,7 @@ void executeOnRegion(DistributedRegionFunctionStreamingMessage msg, final Functi
     final DistributionManager dm = getDistributionManager();
     ResultSender resultSender = new DistributedRegionFunctionResultSender(dm, msg, function);
     final RegionFunctionContextImpl context = new RegionFunctionContextImpl(cache, function.getId(),
-        this, args, filter, null, null, resultSender, isReExecute);
+        this, args, filter, null, null, resultSender, isReExecute, cache.getDistributedSystem());
```

Comment:
We don't need to explicitly pass the DistributedSystem to the context, as this DS is already on the cache